airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py

```diff
@@ -241,8 +244,20 @@ def check(self, logger: logging.Logger, config: Mapping[str, Any]) -> AirbyteCon
             if CONFIG_MOTHERDUCK_API_KEY in config:
                 duckdb_config["motherduck_token"] = str(config[CONFIG_MOTHERDUCK_API_KEY])
                 duckdb_config["custom_user_agent"] = "airbyte"
-
-            con = duckdb.connect(database=path, read_only=False, config=duckdb_config)
+                # Next, we want to specify 'saas_mode' for during check,
+                # to reduce memory usage from unnecessary extensions
+                if "?" in path:
+                    # There are already some query params; append to them.
+                    path += "&saas_mode=true"
+                else:
+                    # No query params yet; add one.
+                    path += "?saas_mode=true"
```

Comment:
@guenp - Can you review the syntax here for `saas_mode`? Somewhat cleaner would be to pass this key-pair as kwargs to connect() or as an entry within `duckdb_config`. The official docs point to appending to the path string, which is what I have here.

https://motherduck.com/docs/key-tasks/authenticating-and-connecting-to-motherduck/authenticating-to-motherduck/#authentication-using-saas-mode

Reply:
This looks good!
It would indeed be cleaner to use `duckdb_config`, but we unfortunately don't support that just yet. I'll open a PR to update this once support for that lands (hopefully next week!)